### ISSUE TITLE
HasDataMapping() doens't work properly

### ISSUE THREAD

**Issue by schaefersimon** [mindsphere/mindconnect-nodejs#147]
Onoarding as well as putting the dataSourceConfiguration and dataMapping to an existing asset worked just fine. Even though a dataMapping existed for the given agent the function HasDataMapping() returned false.

**Comment by sn0wcat** [comment]
HasDataMapping is only returning true if you have created the mappings programmatically. If you want to check if there are data mappings on the mindsphere side you have to call 

```typescript
await agent.GetDataMappings()
```

I will change the documentation to indicate that these two methods are just checking the local agent configuration and not the one on the server.

**Comment by sn0wcat** [comment]
to be fixed in 3.8.0
new documentation
```typescript

/**
 * MindConnect Agent implements the V3 of the Mindsphere API.
 *
 *  * The synchronous methods (IsOnBoarded, HasConfiguration, HasDataMapping...) are operating on agent state storage only.
 *  * The asynchronous methods (GetDataSourceConfiguration, BulkPostData...)  are calling MindSphere APIs.
 *
 * @see https://opensource.mindsphere.io/docs/mindconnect-nodejs/agent-development/agent-state-storage.html
 *
 * @export
 * @class MindConnectAgent
 */
export class MindConnectAgent extends AgentAuth {

 /**
     *
     * Check in the local storage if the agent is onboarded.
     *
     * * This is a local agent state storage setting only. MindSphere API is not called.
     *
     * @see https://opensource.mindsphere.io/docs/mindconnect-nodejs/agent-development/agent-state-storage.html
     *
     * @returns {boolean}
     * @memberof MindConnectAgent
     */
    public IsOnBoarded(): boolean {
        return this._configuration.response ? true : false;
    }

    /**
     * Checks in the local storage if the agent has a data source configuration.
     *
     * * This is a local agent state storage setting only. MindSphere API is not called.
     * * Call await GetDataSourceConfiguration() if you want to check if there is configuration in the mindsphere.
     *
     * @see https://opensource.mindsphere.io/docs/mindconnect-nodejs/agent-development/agent-state-storage.html
     *
     * @returns {boolean}
     * @memberof MindConnectAgent
     */
    public HasDataSourceConfiguration(): boolean {
        if (!this._configuration.dataSourceConfiguration) {
            return false;
        } else if (!this._configuration.dataSourceConfiguration.configurationId) {
            return false;
        } else {
            return true;
        }
    }

    /**
     * Checks in the local storage if the agent has configured mappings.
     *
     * * This is a local agent state storage setting only. MindSphere API is not called.
     * * Call await GetDataMappings() to check if the agent has configured mappings in the MindSphere.
     *
     * @see https://opensource.mindsphere.io/docs/mindconnect-nodejs/agent-development/agent-state-storage.html
     *
     * @returns {boolean}
     * @memberof MindConnectAgent
     */
    public HasDataMappings(): boolean {
        return this._configuration.mappings ? true : false;
    }


}

```
